src/backend/input/tablet.rs

```diff
@@ -0,0 +1,361 @@
+use super::{ButtonState, Event, InputBackend, UnusedEvent};
+
+/// Description of physical tablet tool
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct TabletToolDescriptor {
+    /// The tool type is the high-level type of the tool and usually decides the interaction expected from this tool.
+    pub tool_type: TabletToolType,
+    /// Unique hardware serial number of the tool
+    pub hardware_serial: u64,
+    /// Hardware id in Wacom’s format
+    pub hardware_id_wacom: u64,
+    /// Tool capability
+    /// Notifies the client of any capabilities of this tool, beyond the main set of x/y axes and tip up/down detection
+    pub capabilitys: TabletToolCapabilitys,
+}
+
+/// Describes the physical type of a tool. The physical type of a tool generally defines its base usage.
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+pub enum TabletToolType {
+    /// A generic pen.
+    Pen,
+    /// Eraser
+    Eraser,
+    /// A paintbrush-like tool.
+    Brush,
+    /// Physical drawing tool, e.g. Wacom Inking Pen
+    Pencil,
+    /// An airbrush-like tool.
+    Airbrush,
+    /// A mouse bound to the tablet.
+    Mouse,
+    /// A mouse tool with a lens.
+    Lens,
+    /// A rotary device with positional and rotation data.  
+    Totem,
+}
+
+/// Describes extra capabilities on a tablet.
+///
+/// Any tool must provide x and y values, extra axes are device-specific.
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct TabletToolCapabilitys {
+    /// Tilt axes
+    pub tilt: bool,
+    /// Pressure axis
+    pub pressure: bool,
+    /// Distance axis
+    pub distance: bool,
+    /// Z-rotation axis
+    pub rotation: bool,
+    /// Slider axis
+    pub slider: bool,
+    /// Wheel axis
+    pub wheel: bool,
+}
```

Comment:
Would it be relevant to make this struct into a bitfield, to make it more compact?